### PR TITLE
Utilities for Angular Display system

### PR DESCRIPTION
**Pull request by Leemoonsoo**
### What is this PR for?
[Angular display system](http://zeppelin.incubator.apache.org/docs/0.6.0-incubating-SNAPSHOT/displaysystem/angular.html) provides way to Interpreter interact with front-end.

However, the api was pretty much low level and not very much intuitive, it was difficult to make readable code. This PR provides simple intuitive API for using angular display system by leveraging scala.xml. Following is usage

Import

```scala
// notebook scope
import org.apache.zeppelin.display.angular.notebookscope._
import AngularElem._
```

or

```scala
// paragraph scope
import org.apache.zeppelin.display.angular.paragraphscope._
import AngularElem._
```

Display element

```scala
// automatically convert to string and print with %angular display system directive in front.
<div><div>.display
```

Event handler

```scala
// on click
<div></div>.onClick(() => {
   my callback routine
}).display

// on change
<div></div>.onChange(() => {
  my callback routine
}).display

// arbitrary event
<div></div>.onEvent("ng-click", () => {
  my callback routine
}).display
```

Bind model

```scala
// bind model
<div></div>.model("myModel").display

// bind model with initial value
<div></div>.model("myModel", initialValue).display 
```

Interact with model
```scala
// read model
AngularModel("myModel")()

// update model
AngularModel("myModel", "newValue")
```

### What type of PR is it?
Feature

### How should this be tested?
See usage and screenshot

### Screenshots (if appropriate)

Example of basic usage
image

Example of string converter
display_util

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? documentation will follow

### CODE REVIEW

**Comment by Leemoonsoo** [comment]
Please review this PR

File: zeppelin-display/pom.xml [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+  <groupId>org.apache.zeppelin</groupId>
+  <artifactId>zeppelin-display</artifactId>
+  <packaging>jar</packaging>
+  <version>0.6.0-incubating-SNAPSHOT</version>
+  <name>Zeppelin: Display system apis</name>
+  <url>http://incubator.zeppelin.apache.org</url>
+
+  <properties>
+    <scala.version>2.10.4</scala.version>
```
Comment (felixcheung) [inline zeppelin-display/pom.xml]: shouldn't this be from the parent pom?

**Comment by felixcheung** [comment]
looks good, could you follow up with updating the doc http://zeppelin.incubator.apache.org/docs/0.6.0-incubating-SNAPSHOT/displaysystem/angular.html or similar?

File: zeppelin-display/pom.xml (hunk 1 above)
Comment (Leemoonsoo) [inline zeppelin-display/pom.xml]: Zeppelin submodule can have different scala.version dependency, especially for interpreter. While one interpreter uses scala 2.10, the other uses scala 2.11. So i think it's better not having scala.version in root pom file.

Comment (felixcheung) [inline zeppelin-display/pom.xml]: Sure, it makes sense. Wouldn't the display module running in the Spark interpreter process and would need to match scala build to work?

**Comment by doanduyhai** [comment]
@Leemoonsoo Can you provide a screen capture/code sample to show how we can use those utilities function to update an Angular object server-side from the front-end ?

In all the above screenshots, what is the interpreter in use ? %angular or %spark ?

File: zeppelin-display/pom.xml (hunk 1 above)
Comment (Leemoonsoo) [inline zeppelin-display/pom.xml]: Right, If there're interpreter that uses scala 2.11 for example, then display module may add support for scala 2.11.

**Comment by Leemoonsoo** [comment]
@doanduyhai The examples in this PR's description can be run on %spark interpreter.
Currently, push new angular object from front-end is not possible. I think that's useful feature to have and can be handled in the different PR, while this PR trying to provide some convenient utility function for existing angular object.

**Comment by doanduyhai** [comment]
Thanks for the clarifications @Leemoonsoo. I'll try to push a small PR for the front-end angular value push

**Comment by Leemoonsoo** [comment]
I've made small change after https://github.com/apache/incubator-zeppelin/pull/588 merged.
Now it provides two packages `org.apache.zeppelin.display.angular.notebookscope` and `org.apache.zeppelin.display.angular.paragraphscope`. One work in notebook scope and the other work in paragraph scope.

Usage now slightly changed, when import these utilities.

When you want to work in notebook scope
```scala
import org.apache.zeppelin.display.angular.notebookscope._
import AngularElem._
```

When you want to work in paragraph scope
```scala
import org.apache.zeppelin.display.angular.paragraphscope._
import AngularElem._
```

**Comment by corneadoug** [comment]
#650 needs to be updated too then

**Comment by AhyoungRyu** [comment]
#650 is updated.

**Comment by Leemoonsoo** [comment]
Merge #591 and #650 if there're no more discussions.

**Pull request closed without merging**